tests/test_packages/test_protocols/test_fipa.py

```diff
@@ -290,16 +290,203 @@ def test_fipa_decoding_unknown_performative():
+                          info={"address": "dummy_address"})
+        msg.counterparty = "opponent"
+        response = self.dialogues.is_belonging_to_registered_dialogue(msg, agent_addr="starter")
+        assert response, "We expect the response from the function to be true."
+
+    def test_get_dialogues_other_initiated(self):
+        """Test the returned opponent initiated dialogues."""
+        opponent_result = self.dialogues.create_opponent_initiated(dialogue_opponent_addr="opponent",
+                                                                   dialogue_reference=(str(0), ''), is_seller=True)
+        msg = FIPAMessage(message_id=1,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                          info={"address": "dummy_address"})
+
+        msg.counterparty = "opponent"
+        opponent_initiated_dialogue_label = DialogueLabel((str(0), ''), "opponent", "opponent")
+        self.dialogues.dialogues[opponent_initiated_dialogue_label] = cast(FIPADialogue, opponent_result)
+        opponent_retrieved_dialogue = self.dialogues.get_dialogue(fipa_msg=msg, agent_addr="starter")
+        assert opponent_retrieved_dialogue.is_self_initiated
+
+    def test_get_dialogues_self_initiated(self):
+        """Test the returned self initiated dialogues."""
+        result = self.dialogues.create_self_initiated(dialogue_starter_addr="starter", dialogue_opponent_addr="opponent",
+                                                      is_seller=True)
+        result._outgoing_messages.append(self.last_msg)
+        self_initiated_dialogue_label = DialogueLabel((str(0), ''), "opponent", "starter")
+        self.dialogues.dialogues[self_initiated_dialogue_label] = cast(FIPADialogue, result)
+        msg = FIPAMessage(message_id=1,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                          info={"address": "dummy_address"})
+        msg.counterparty = "opponent"
+        retrieved_dialogue = self.dialogues.get_dialogue(fipa_msg=msg, agent_addr="starter")
+        assert retrieved_dialogue.is_self_initiated
+
+    def test_get_dialogues_value_error(self):
+        """Test the value error of the get dialogues function."""
+        msg = FIPAMessage(message_id=1,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                          info={"address": "dummy_address"})
+        msg.counterparty = "opponent"
+        with pytest.raises(ValueError, match="Should have found dialogue."):
+            self.dialogues.get_dialogue(fipa_msg=msg, agent_addr="unknown_addr")
+
+    def test_add_dialogue(self):
```

Comment:
The add functionality should be tested implicitly by one of the above test cases, let's discuss